docs/TPM2_PCR_MEASUREMENTS.md

```diff
@@ -0,0 +1,137 @@
+---
+title: Recognizing TPM2 PCR Measurements Made by systemd
+category: Booting
+layout: default
+SPDX-License-Identifier: LGPL-2.1-or-later
+---
+
+# Recognizing TPM2 PCR Measurements Made by systemd
+
+Various systemd components issue TPM2 PCR measurements during the boot process,
+both in UEFI mode and from userspace. The following lists all measurements
+done, and describes (in case done before `ExitBootServices()`) how they appear
+in the TPM2 Event Log, maintained by the PC firmware.
+
+systemd will measure to PCRs 11, 12, 13, 15.
+
+Note that the userspace measurements listed below are only done on systems
+booted with `systemd-stub`.
+
+## PCR Measurements Made by `systemd-boot` (UEFI)
+
+### PCR 12, EV_IPL, "Kernel Command Line"
+
+If the kernel command line was specified explicitly (by the user or in a Boot
+Loader Specification Type #1 file), the kernel command line passed to the
+invoked kernel is measured before it is executed.
+
+*Description* in the event log record is the literal kernel command line in
+UTF-16.
+
+*Measured hash* covers the literal kernel command line in UTF-16 (without any
+trailing NUL byte).
+
+## PCR Measurements Made by `systemd-stub` (UEFI)
+
+### PCR 11, EV_IPL "PE Section Name"
+
+Happens once for each UKI-defined PE section of the UKI, in the canonical UKI
```

Comment:
"A measurement is made for each PE section of the UKI that is described by the UKI specification (link), in the canonical order described in the specification."